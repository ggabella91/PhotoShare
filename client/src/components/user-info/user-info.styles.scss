.user-suggestion-container {
  display: flex;
  flex-direction: column;
  position: absolute;
  width: 220px;
  max-height: 100px;
  background-color: white;
  overflow-y: scroll;
  overflow-x: hidden;
  border-radius: 4px;
  transform: translate(-10px);
  z-index: 20;

  .user-suggestion-element {
    display: flex;
    position: relative;
    justify-content: start;
    width: 220px;
    height: 50px;
    background-color: white;
    border-bottom: solid 1px rgb(180, 180, 180);

    &:hover {
      cursor: pointer;
    }

    .suggestion-avatar {
      position: relative;
      width: 50px;
      height: 50px;
      border-radius: 50%;

      .suggestion-profile-photo {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 32px;
        height: 32px;
        border-radius: 50%;
      }

      .suggestion-photo-placeholder {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 32px;
        height: 32px;
        border-radius: 50%;
        border: solid 1px;
        border-color: black;
        display: flex;
        flex-direction: column;
        align-items: center;

        .suggestion-photo-placeholder-text {
          position: relative;
          font-size: 8px;
          color: black;
          margin-top: 50%;
          transform: translateY(-55%);
        }
      }
    }

    .suggestion-username-and-other-data {
      display: flex;
      position: relative;
      flex-direction: column;
      padding-top: 4px;
      width: 140px;
      height: 50px;
      justify-self: flex-end;
      text-align: left;

      .suggestion-username {
        display: flex;
        position: relative;
        font-size: 14px;
        font-weight: 420;
        color: black;
      }

      .suggestion-name {
        display: flex;
        position: relative;
        font-size: 12px;
        font-weight: 380;
        color: black;
      }

      .suggestion-location {
        display: none;
      }
    }
  }
}

.user-modal-container {
  display: flex;
  flex-direction: column;
  position: relative;
  width: inherit;
  max-height: 338px;
  background-color: white;
  border-radius: 4px;

  .user-modal-element {
    display: flex;
    position: relative;
    justify-content: start;
    width: 100%;
    height: 70px;
    background-color: white;
    border-bottom: solid 1px rgb(180, 180, 180);
    padding-top: 5px;
    padding-bottom: 5px;

    &:hover {
      cursor: pointer;
    }

    .modal-avatar {
      position: relative;
      width: 60px;
      height: 60px;
      border-radius: 50%;
      margin-left: 15px;
      margin-right: 2vw;

      .modal-profile-photo {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 45px;
        height: 45px;
        border-radius: 50%;
      }

      .modal-photo-placeholder {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 45px;
        height: 45px;
        border-radius: 50%;
        border: solid 1px;
        border-color: black;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        .modal-photo-placeholder-text {
          position: relative;
          font-size: 10px;
          color: black;
          margin-top: 50%;
          margin-left: 50%;
          transform: translate(-50%, -50%);
        }
      }
    }

    .modal-username-and-other-data {
      display: flex;
      position: relative;
      flex-direction: column;
      padding-top: 6px;
      width: 75%;
      height: 60px;
      justify-self: flex-end;
      text-align: left;

      .modal-username {
        display: flex;
        position: relative;
        font-size: 16px;
        font-weight: 420;
        color: black;
      }

      .modal-name {
        display: flex;
        position: relative;
        font-size: 14px;
        font-weight: 380;
        color: black;
      }

      .modal-location {
        display: none;
      }
    }
  }
}

.user-feed-container {
  display: flex;
  flex-direction: column;
  position: relative;
  width: inherit;
  max-height: 338px;
  background-color: white;
  border-radius: 4px;

  .user-feed-element {
    display: flex;
    position: relative;
    justify-content: start;
    width: 100%;
    height: 60px;
    background-color: white;
    border-bottom: solid 1px rgb(180, 180, 180);
    padding-top: 5px;
    padding-bottom: 5px;

    &:hover {
      cursor: pointer;
    }

    .feed-avatar {
      position: relative;
      width: 50px;
      height: 50px;
      border-radius: 50%;
      margin-left: 15px;
      margin-right: 15px;
      padding-top: 5px;

      .feed-profile-photo {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 40px;
        height: 40px;
        border-radius: 50%;
      }

      .feed-photo-placeholder {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 40px;
        height: 40px;
        border-radius: 50%;
        border: solid 1px;
        border-color: black;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        .feed-photo-placeholder-text {
          position: relative;
          font-size: 10px;
          color: black;
          margin-top: 50%;
          margin-left: 50%;
          transform: translate(-50%, -50%);
        }
      }
    }

    .feed-username-and-other-data {
      display: flex;
      position: relative;
      flex-direction: column;
      padding-top: 6px;
      width: 75%;
      height: 60px;
      justify-self: flex-end;
      text-align: left;

      .feed-username {
        display: flex;
        position: relative;
        font-size: 14px;
        font-weight: 540;
        color: black;
      }

      .feed-name {
        display: none;
      }

      .feed-location {
        display: flex;
        position: relative;
        font-size: 12px;
        font-weight: 380;
        color: black;
      }
    }
  }
}

.user-comment-container {
  display: flex;
  flex-direction: column;
  position: relative;
  width: inherit;
  max-height: 338px;
  background-color: white;
  border-radius: 4px;

  .user-comment-element {
    display: flex;
    position: relative;
    justify-content: start;
    width: 100%;
    height: 60px;
    background-color: white;
    border-bottom: solid 1px rgb(180, 180, 180);
    padding-top: 5px;
    padding-bottom: 5px;

    &:hover {
      cursor: pointer;
    }

    .comment-avatar {
      position: relative;
      width: 50px;
      height: 50px;
      border-radius: 50%;
      margin-left: 15px;
      margin-right: 15px;
      padding-top: 5px;

      .comment-profile-photo {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 40px;
        height: 40px;
        border-radius: 50%;
      }

      .comment-photo-placeholder {
        position: relative;
        margin-left: 50%;
        margin-top: 50%;
        transform: translate(-50%, -50%);
        width: 40px;
        height: 40px;
        border-radius: 50%;
        border: solid 1px;
        border-color: black;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        .comment-photo-placeholder-text {
          position: relative;
          font-size: 10px;
          color: black;
          margin-top: 50%;
          margin-left: 50%;
          transform: translate(-50%, -50%);
        }
      }
    }

    .comment-username-and-other-data {
      display: flex;
      position: relative;
      flex-direction: column;
      padding-top: 6px;
      width: 75%;
      height: 60px;
      justify-self: flex-end;
      text-align: left;

      .comment-username {
        display: flex;
        position: relative;
        font-size: 14px;
        font-weight: 540;
        color: black;
      }

      .comment-name {
        display: none;
      }

      .comment-location {
        display: none;
      }

      .comment-text {
        display: flex;
        position: relative;
        font-size: 12;
        font-weight: 400;
      }
    }

    .comment-date {
      display: flex;
      position: relative;
      font-size: 10;
      font-weight: 380;
      margin-top: 5px;
    }
  }
}
