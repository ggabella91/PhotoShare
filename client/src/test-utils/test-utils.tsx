import { FC, ReactElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { store } from '../redux/store';

interface WrapperProps {
  wrapperProps?: { route: string; location: string };
}

type CustomRenderOptions = RenderOptions & WrapperProps;

const WithReduxProvider: FC<WrapperProps> = ({ children, wrapperProps }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path={wrapperProps?.route || '/'} element={children} />
        </Routes>
      </BrowserRouter>
    </Provider>
  );
};

const renderWithReduxAndRouter = (
  ui: ReactElement,
  options?: Omit<CustomRenderOptions, 'wrapper'>
) => {
  window.history.pushState({}, '', options?.wrapperProps?.location || '/');
  const customContainer = document.createElement('div');
  customContainer.id = 'root';

  return render(ui, {
    wrapper: (props) => (
      <WithReduxProvider {...props} wrapperProps={options?.wrapperProps} />
    ),
    container: document.body.appendChild(customContainer),
    ...options,
  });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { renderWithReduxAndRouter as render };
