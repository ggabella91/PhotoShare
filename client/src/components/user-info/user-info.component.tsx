import React, { useState } from 'react';
import { useHistory, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { createStructuredSelector } from 'reselect';

import { AppState } from '../../redux/root-reducer';

import { DeleteReactionReq } from '../../redux/post/post.types';

import {
  setCommentToDelete,
  setShowCommentOptionsModal,
  setShowPostEditForm,
} from '../../redux/post/post.actions';

import './user-info.styles.scss';

export enum StyleType {
  suggestion = 'suggestion',
  modal = 'modal',
  feed = 'feed',
  comment = 'comment',
}

export interface UserInfoAndOtherData {
  profilePhotoFileString: string;
  username: string;
  name: string;
  location: string;
  comment: string;
  commentDate?: Date | string;
  reactionId?: string;
  reactingUserId?: string;
}

interface UserInfoProps {
  styleType: StyleType;
  userInfoArray: UserInfoAndOtherData[];
  isCaption?: boolean;
  isCaptionOwner?: boolean;
  setCommentToDelete: typeof setCommentToDelete;
  setShowCommentOptionsModal: typeof setShowCommentOptionsModal;
  setShowPostEditForm: typeof setShowPostEditForm;
}

export const UserInfo: React.FC<UserInfoProps> = ({
  userInfoArray,
  styleType,
  isCaption,
  isCaptionOwner,
  setCommentToDelete,
  setShowCommentOptionsModal,
  setShowPostEditForm,
}) => {
  const [showCommentOptionsButtonForIdx, setShowCommentOptionsButtonForIdx] =
    useState({ show: false, idx: -1 });

  let history = useHistory();

  const handleSetCommentToDelete = (idx: number) => {
    const commentToDelete = userInfoArray[idx];
    if (commentToDelete.reactionId && commentToDelete.reactingUserId) {
      setCommentToDelete({
        reactionId: commentToDelete.reactionId,
        isLikeRemoval: false,
        reactingUserId: commentToDelete.reactingUserId,
      });

      setShowCommentOptionsModal(true);
    }
  };

  const handleClickCaptionOptions = () => {
    if (isCaptionOwner) {
      setShowPostEditForm(true);
    }
  };

  const userInfo = userInfoArray.map(
    (el: UserInfoAndOtherData, idx: number) => (
      <div
        className={`user-${styleType}-element`}
        key={idx}
        onClick={
          styleType === StyleType.suggestion
            ? () => {
                history.push(`/${el.username}`);
              }
            : () => {}
        }
        onMouseEnter={() =>
          setShowCommentOptionsButtonForIdx({ show: true, idx })
        }
        onMouseLeave={() =>
          setShowCommentOptionsButtonForIdx({ show: false, idx })
        }
      >
        <div className={`${styleType}-avatar`}>
          {el.profilePhotoFileString ? (
            <img
              className={`${styleType}-profile-photo`}
              src={`data:image/jpeg;base64,${el.profilePhotoFileString}`}
              alt='profile-pic'
            />
          ) : null}
          {!el.profilePhotoFileString ? (
            <div className={`${styleType}-photo-placeholder`}>
              <span className={`${styleType}-photo-placeholder-text`}>
                No photo
              </span>
            </div>
          ) : null}
        </div>
        <div className='user-data-and-date'>
          <div className={`${styleType}-username-and-other-data`}>
            {styleType !== StyleType.suggestion ? (
              <NavLink
                to={`/${el.username}`}
                className={`${styleType}-username`}
              >
                {el.username}
              </NavLink>
            ) : (
              <span className={`${styleType}-username`}>{el.username}</span>
            )}
            <span className={`${styleType}-name`}>{el.name}</span>
            <span className={`${styleType}-location`}>{el.location}</span>
            <span>{el.comment ? el.comment : null}</span>
          </div>
          {el.commentDate ? (
            <span className={`${styleType}-date`}>
              {new Date(el.commentDate).toDateString()}
            </span>
          ) : null}
        </div>
        <div
          className={`${
            showCommentOptionsButtonForIdx.show &&
            showCommentOptionsButtonForIdx.idx === idx
              ? ''
              : 'hide'
          } comment-options`}
        >
          {styleType === StyleType.comment && !isCaption ? (
            <span
              className='comment-ellipsis-button'
              onClick={() => handleSetCommentToDelete(idx)}
            >
              ...
            </span>
          ) : null}
          {styleType === StyleType.comment && isCaption ? (
            <span
              className='comment-ellipsis-button'
              onClick={handleClickCaptionOptions}
            >
              ...
            </span>
          ) : null}
        </div>
      </div>
    )
  );

  return <div className={`user-${styleType}-container`}>{userInfo}</div>;
};

interface LinkStateProps {}

const mapStateToProps = createStructuredSelector<AppState, LinkStateProps>({});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  setCommentToDelete: (deleteReactionReq: DeleteReactionReq) =>
    dispatch(setCommentToDelete(deleteReactionReq)),
  setShowCommentOptionsModal: (showCommentOptionsModal: boolean) =>
    dispatch(setShowCommentOptionsModal(showCommentOptionsModal)),
  setShowPostEditForm: (showPostEditForm: boolean) =>
    dispatch(setShowPostEditForm(showPostEditForm)),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserInfo);
