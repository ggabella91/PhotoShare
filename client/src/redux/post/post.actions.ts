import { UserInfoAndOtherData } from '../../components/user-info/user-info.component';
import { PostModalDataToFeed } from '../../components/feed-post-container/feed-post-container.component';

import {
  Post,
  PostDataReq,
  PostFileReq,
  Reaction,
  ReactionReq,
  ArchivePostReq,
  DeleteReactionReq,
  PostFile,
  PostError,
  PostActions,
  PostActionTypes,
  PostMetaData,
} from './post.types';

export const createPostStart = (post: FormData): PostActionTypes => ({
  type: PostActions.CREATE_POST_START,
  payload: post,
});

export const createPostSuccess = (post: Post): PostActionTypes => ({
  type: PostActions.CREATE_POST_SUCCESS,
  payload: post,
});

export const createPostFailure = (error: PostError): PostActionTypes => ({
  type: PostActions.CREATE_POST_FAILURE,
  payload: error,
});

export const createPostReactionStart = (
  reactionReq: ReactionReq
): PostActionTypes => ({
  type: PostActions.CREATE_POST_REACTION_START,
  payload: reactionReq,
});

export const createPostReactionSuccess = (
  reaction: Reaction
): PostActionTypes => ({
  type: PostActions.CREATE_POST_REACTION_SUCCESS,
  payload: reaction,
});

export const createPostReactionFailure = (
  error: PostError
): PostActionTypes => ({
  type: PostActions.CREATE_POST_REACTION_FAILURE,
  payload: error,
});

export const getPostReactionsStart = (postId: string): PostActionTypes => ({
  type: PostActions.GET_POST_REACTIONS_START,
  payload: postId,
});

export const getPostReactionsSuccess = (
  reactions: Reaction[]
): PostActionTypes => ({
  type: PostActions.GET_POST_REACTIONS_SUCCESS,
  payload: reactions,
});

export const getPostReactionsFailure = (error: PostError): PostActionTypes => ({
  type: PostActions.GET_POST_REACTIONS_FAILURE,
  payload: error,
});

export const clearPostStatuses = (): PostActionTypes => ({
  type: PostActions.CLEAR_POST_STATUSES,
  payload: null,
});

export const updateProfilePhotoStart = (photo: FormData): PostActionTypes => ({
  type: PostActions.UPDATE_PROFILE_PHOTO_START,
  payload: photo,
});

export const updateProfilePhotoSuccess = (photo: Post): PostActionTypes => ({
  type: PostActions.UPDATE_PROFILE_PHOTO_SUCCESS,
  payload: photo,
});

export const updateProfilePhotoFailure = (
  error: PostError
): PostActionTypes => ({
  type: PostActions.UPDATE_PROFILE_PHOTO_FAILURE,
  payload: error,
});

export const clearProfilePhotoStatuses = (): PostActionTypes => ({
  type: PostActions.CLEAR_PROFILE_PHOTO_STATUSES,
  payload: null,
});

export const getPostDataStart = (
  postDataReq: PostDataReq
): PostActionTypes => ({
  type: PostActions.GET_POST_DATA_START,
  payload: postDataReq,
});

export const getPostDataSuccess = (postDataArray: Post[]): PostActionTypes => ({
  type: PostActions.GET_POST_DATA_SUCCESS,
  payload: postDataArray,
});

export const addPostDataToFeedArray = (
  postDataArray: Post[]
): PostActionTypes => ({
  type: PostActions.ADD_POST_DATA_TO_FEED_ARRAY,
  payload: postDataArray,
});

export const getPostDataFailure = (error: PostError): PostActionTypes => ({
  type: PostActions.GET_POST_DATA_FAILURE,
  payload: error,
});

export const getPostFileStart = (fileReq: PostFileReq): PostActionTypes => ({
  type: PostActions.GET_POST_FILE_START,
  payload: fileReq,
});

export const getPostFileSuccess = (file: PostFile): PostActionTypes => ({
  type: PostActions.GET_POST_FILE_SUCCESS,
  payload: file,
});

export const getPostFileFailure = (error: PostError): PostActionTypes => ({
  type: PostActions.GET_POST_FILE_FAILURE,
  payload: error,
});

export const getProfilePhotoFileSuccess = (
  file: PostFile
): PostActionTypes => ({
  type: PostActions.GET_PROFILE_PHOTO_FILE_SUCCESS,
  payload: file,
});

export const archivePostStart = (
  archiveReq: ArchivePostReq
): PostActionTypes => ({
  type: PostActions.ARCHIVE_POST_START,
  payload: archiveReq,
});

export const archivePostSuccess = (message: string): PostActionTypes => ({
  type: PostActions.ARCHIVE_POST_SUCCESS,
  payload: message,
});

export const archivePostFailure = (error: Error): PostActionTypes => ({
  type: PostActions.ARCHIVE_POST_FAILURE,
  payload: error,
});

export const deleteReactionStart = (
  deleteReactionReq: DeleteReactionReq
): PostActionTypes => ({
  type: PostActions.DELETE_REACTION_START,
  payload: deleteReactionReq,
});

export const deleteReactionSuccess = (message: string): PostActionTypes => ({
  type: PostActions.DELETE_REACTION_SUCCESS,
  payload: message,
});

export const deleteReactionFailure = (error: PostError): PostActionTypes => ({
  type: PostActions.DELETE_REACTION_FAILURE,
  payload: error,
});

export const getOtherUserProfilePhotoFileSuccess = (
  file: string
): PostActionTypes => ({
  type: PostActions.GET_OTHER_USER_PROFILE_PHOTO_FILE_SUCCESS,
  payload: file,
});

export const getUserPhotoForFollowArraySuccess = (
  postFile: PostFile
): PostActionTypes => ({
  type: PostActions.GET_USER_PHOTO_FOR_FOLLOW_ARRAY_SUCCESS,
  payload: postFile,
});

export const getUserPhotoForSuggestionArraySuccess = (
  postFile: PostFile
): PostActionTypes => ({
  type: PostActions.GET_USER_PHOTO_FOR_SUGGESTION_ARRAY_SUCCESS,
  payload: postFile,
});

export const getUserPhotoForReactorArraySuccess = (
  postFile: PostFile
): PostActionTypes => ({
  type: PostActions.GET_USER_PHOTO_FOR_REACTOR_ARRAY_SUCCESS,
  payload: postFile,
});

export const clearArchivePostStatuses = (): PostActionTypes => ({
  type: PostActions.CLEAR_ARCHIVE_POST_STATUSES,
  payload: null,
});

export const clearFollowPhotoFileArray = (): PostActionTypes => ({
  type: PostActions.CLEAR_FOLLOW_PHOTO_FILE_ARRAY,
  payload: null,
});

export const clearSuggestionPhotoFileArray = (): PostActionTypes => ({
  type: PostActions.CLEAR_SUGGESTION_PHOTO_FILE_ARRAY,
  payload: null,
});

export const clearPostFilesAndData = (): PostActionTypes => ({
  type: PostActions.CLEAR_POST_FILES_AND_DATA,
  payload: null,
});

export const clearPostState = (): PostActionTypes => ({
  type: PostActions.CLEAR_POST_STATE,
  payload: null,
});

export const setCommentToDelete = (
  deleteReactionReq: DeleteReactionReq
): PostActionTypes => ({
  type: PostActions.SET_COMMENT_TO_DELETE,
  payload: deleteReactionReq,
});

export const setShowCommentOptionsModal = (
  showCommentOptionsModal: boolean
): PostActionTypes => ({
  type: PostActions.SET_SHOW_COMMENT_OPTIONS_MODAL,
  payload: showCommentOptionsModal,
});

export const clearPostReactions = (): PostActionTypes => ({
  type: PostActions.CLEAR_POST_REACTIONS,
  payload: null,
});

export const setPostMetaDataForUser = (
  postMetaData: PostMetaData
): PostActionTypes => ({
  type: PostActions.SET_POST_META_DATA_FOR_USER,
  payload: postMetaData,
});

export const setPostLikingUsersArray = (
  postLikingUsersArray: UserInfoAndOtherData[]
): PostActionTypes => ({
  type: PostActions.SET_POST_LIKING_USERS_ARRAY,
  payload: postLikingUsersArray,
});

export const setShowPostLikingUsersModal = (
  showPostLikingUsersModal: boolean
): PostActionTypes => ({
  type: PostActions.SET_SHOW_POST_LIKING_USERS_MODAL,
  payload: showPostLikingUsersModal,
});

export const setFeedPagePostModalData = (
  postModalDataToFeed: PostModalDataToFeed
): PostActionTypes => ({
  type: PostActions.SET_FEED_PAGE_POST_MODAL_DATA,
  payload: postModalDataToFeed,
});

export const setFeedPagePostModalShow = (
  postModalDataShow: boolean
): PostActionTypes => ({
  type: PostActions.SET_FEED_PAGE_POST_MODAL_SHOW,
  payload: postModalDataShow,
});

export const setFeedPagePostOptionsModalShow = (
  postOptionsModalDataShow: boolean
): PostActionTypes => ({
  type: PostActions.SET_FEED_PAGE_POST_OPTIONS_MODAL_SHOW,
  payload: postOptionsModalDataShow,
});

export const setClearFeedPagePostModalState = (
  clearFeedPagePostModalState: boolean
): PostActionTypes => ({
  type: PostActions.SET_CLEAR_FEED_PAGE_POST_MODAL_STATE,
  payload: clearFeedPagePostModalState,
});

// Actions for feed-post-container

export const getFeedPostFileSuccess = (file: PostFile): PostActionTypes => ({
  type: PostActions.GET_FEED_POST_FILE_SUCCESS,
  payload: file,
});

export const getFeedPostReactionsSuccess = (
  reactions: Reaction[]
): PostActionTypes => ({
  type: PostActions.GET_FEED_POST_REACTIONS_SUCCESS,
  payload: reactions,
});

export const getUserPhotoForFeedReactorArraySuccess = (
  postFile: PostFile
): PostActionTypes => ({
  type: PostActions.GET_USER_PHOTO_FOR_FEED_REACTOR_ARRAY_SUCCESS,
  payload: postFile,
});
