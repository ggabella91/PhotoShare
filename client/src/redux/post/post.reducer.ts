import { PostActions, PostActionTypes, PostState } from './post.types';

const INITIAL_STATE: PostState = {
  postData: null,
  postFiles: [],
  getPostDataError: null,
  getPostDataConfirm: null,
  getPostFileError: null,
  getPostFileConfirm: null,
  postError: null,
  postConfirm: null,
  profilePhotoFile: null,
  profilePhotoError: null,
  profilePhotoConfirm: null,
};

const postReducer = (state = INITIAL_STATE, action: PostActionTypes) => {
  switch (action.type) {
    case PostActions.UPDATE_PROFILE_PHOTO_SUCCESS:
      return {
        ...state,
        profilePhotoError: null,
        profilePhotoConfirm: 'Profile photo updated!',
      };
    case PostActions.CREATE_POST_SUCCESS:
      return {
        ...state,
        postError: null,
        postConfirm: 'Post uploaded!',
      };
    case PostActions.GET_POST_DATA_SUCCESS:
      return {
        ...state,
        postData: action.payload,
        getPostDataError: null,
        getPostDataConfirm: 'Post data fetched!',
      };
    case PostActions.GET_POST_FILE_SUCCESS:
      return {
        ...state,
        postFiles: [...state.postFiles, action.payload],
        getPostFileError: null,
        getPostFileConfirm: 'Post file fetched!',
      };
    case PostActions.GET_PROFILE_PHOTO_FILE_SUCCESS:
      return {
        ...state,
        profilePhotoFile: action.payload,
        getPostFileError: null,
        getPostFileConfirm: 'Profile photo file fetched!',
      };
    case PostActions.UPDATE_PROFILE_PHOTO_FAILURE:
      return {
        ...state,
        profilePhotoError: action.payload,
        profilePhotoConfirm: null,
      };
    case PostActions.CREATE_POST_FAILURE:
      return {
        ...state,
        postError: action.payload,
        postConfirm: null,
      };
    case PostActions.GET_POST_DATA_FAILURE:
      return {
        ...state,
        getPostDataError: action.payload,
        getPostDataConfirm: null,
      };
    case PostActions.GET_POST_FILE_FAILURE:
      return {
        ...state,
        getPostFileError: action.payload,
        getPostFileConfirm: null,
      };
    default:
      return state;
  }
};

export default postReducer;
