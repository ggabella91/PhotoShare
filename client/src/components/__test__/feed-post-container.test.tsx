import { render, screen } from '../../test-utils/test-utils';
import {
  FeedPostContainer,
  PostModalDataToFeed,
} from '../feed-post-container/feed-post-container.component';

import { DeleteReactionReq, PostFileReq } from '../../redux/post/post.types';
import {
  createPostReactionStart,
  getPostReactionsStart,
  deleteReactionStart,
  getPostFileStart,
  setPostLikingUsersArray,
  setFeedPagePostModalData,
  setFeedPagePostModalShow,
  setClearFeedPagePostModalState,
  setShowPostLikingUsersModal,
  clearPostReactions,
} from '../../redux/post/post.actions';
import { OtherUserRequest } from '../../redux/user/user.types';
import { getOtherUserStart } from '../../redux/user/user.actions';
import { UserInfoAndOtherData } from '../user-info/user-info.component';

describe('feed post container component tests', () => {
  const setup = () =>
    render(
      <FeedPostContainer
        s3Key='post-key'
        custRef={null}
        key='0'
        currentUser={{
          name: 'test name',
          username: 'testname',
          id: '2323',
          email: 'test@test.com',
        }}
        caption='test caption'
        userInfo={{
          profilePhotoFileString: 'sndfjnbss',
          username: 'test-user',
          location: 'Bali',
          name: 'Test User',
          userId: 'user-id',
          postId: 'post-id',
          comment: 'comment',
        }}
        fileString='string'
        date='March 19th 2021'
        feedPostReactionsArray={[]}
        getOtherUserStart={(otherUserReq: OtherUserRequest) =>
          getOtherUserStart(otherUserReq)
        }
        getPostFileStart={(postFileReq: PostFileReq) =>
          getPostFileStart(postFileReq)
        }
        feedPostReactingUsers={[]}
        reactorPhotoFileArray={[]}
        usersProfilePhotoConfirm=''
        createPostReactionStart={(reactionReq) =>
          createPostReactionStart(reactionReq)
        }
        deleteReactionStart={(deleteReactionReq: DeleteReactionReq) =>
          deleteReactionStart(deleteReactionReq)
        }
        deleteReactionConfirm={{ reactionId: '', postId: '', message: '' }}
        postReactionConfirm={{
          reactionId: '',
          message: '',
          likedPost: false,
          postId: '',
        }}
        postReactionError={null}
        getPostReactionsStart={(postId) => getPostReactionsStart(postId)}
        getPostReactionsConfirm=''
        getPostReactionsError={null}
        setPostLikingUsersArray={(
          postLikingUsersArray: UserInfoAndOtherData[]
        ) => setPostLikingUsersArray(postLikingUsersArray)}
        setShowPostLikingUsersModal={(showPostLikingUsersModal: boolean) =>
          setShowPostLikingUsersModal(showPostLikingUsersModal)
        }
        setFeedPagePostModalData={(postModalDataToFeed: PostModalDataToFeed) =>
          setFeedPagePostModalData(postModalDataToFeed)
        }
        setFeedPagePostModalShow={(feedPagePostModalShow: boolean) =>
          setFeedPagePostModalShow(feedPagePostModalShow)
        }
        setClearFeedPagePostModalState={(
          clearFeedPagePostModalState: boolean
        ) => setClearFeedPagePostModalState(clearFeedPagePostModalState)}
        clearPostReactions={() => clearPostReactions()}
      />
    );

  it('expect to render a feed post container component', () => {
    setup();

    const feedPostContainer = screen.getByTestId('feed-post-container');

    expect(feedPostContainer).toBeInTheDocument();
  });

  it('expect submit-comment button to be disabled when no comment has been entered', () => {
    setup();

    const submitButton = screen.getByTestId('submit-comment-button');

    expect(submitButton).toBeDisabled();
  });
});
