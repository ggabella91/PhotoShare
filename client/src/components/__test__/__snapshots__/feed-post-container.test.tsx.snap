// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render a feed post container component 1`] = `
<div
  className="feed-post-container"
>
  <div
    className="profile-and-options"
  >
    <UserInfo
      styleType="feed"
      userInfoArray={
        Array [
          Object {
            "location": "Bali",
            "name": "Test User",
            "profilePhotoFileString": "sndfjnbss",
            "username": "test-user",
          },
        ]
      }
    />
  </div>
  <img
    alt="user"
    className="feed-post-photo"
    src="data:image/jpeg;base64,string"
  />
  <div
    className="caption-and-reactions"
  >
    <div
      className="caption"
    >
      <span
        className="username"
      >
        test-user
      </span>
       
      test caption
    </div>
    <span
      className="date"
    >
      March 19th 2021
    </span>
  </div>
</div>
`;
